common/persistence/client/fault_injection.go

```diff
@@ -0,0 +1,925 @@
+	if err := e.errorGenerator.Generate(); err != nil {
+		return err
+	}
+	return e.baseExecutionStore.ForkHistoryBranch(request)
+}
+
+func (e *FaultInjectionExecutionStore) DeleteHistoryBranch(request *persistence.InternalDeleteHistoryBranchRequest) error {
+	if err := e.errorGenerator.Generate(); err != nil {
+		return err
+	}
+	return e.baseExecutionStore.DeleteHistoryBranch(request)
+}
+
+func (e *FaultInjectionExecutionStore) GetHistoryTree(request *persistence.GetHistoryTreeRequest) (
+	*persistence.InternalGetHistoryTreeResponse,
+	error,
+) {
+	if err := e.errorGenerator.Generate(); err != nil {
+		return nil, err
+	}
+	return e.baseExecutionStore.GetHistoryTree(request)
+}
+
+func (e *FaultInjectionExecutionStore) GetAllHistoryTreeBranches(request *persistence.GetAllHistoryTreeBranchesRequest) (
+	*persistence.InternalGetAllHistoryTreeBranchesResponse,
+	error,
+) {
+	if err := e.errorGenerator.Generate(); err != nil {
+		return nil, err
+	}
+	return e.baseExecutionStore.GetAllHistoryTreeBranches(request)
+}
+
+func NewFaultInjectionClusterMetadataStore(config *config.FaultInjection, baseStore persistence.ClusterMetadataStore) (
+	persistence.ClusterMetadataStore,
+	error,
+) {
+	errorGenerator := newErrorGenerator(config)
+	return &FaultInjectionClusterMetadataStore{
+		baseCMStore:    baseStore,
+		config:         config,
+		errorGenerator: errorGenerator,
+	}, nil
+}
+
+func (c *FaultInjectionClusterMetadataStore) Close() {
+	c.baseCMStore.Close()
+}
+
+func (c *FaultInjectionClusterMetadataStore) GetName() string {
+	return c.baseCMStore.GetName()
+}
+
+func (c *FaultInjectionClusterMetadataStore) GetClusterMetadata() (
+	*persistence.InternalGetClusterMetadataResponse,
+	error,
+) {
+	if err := c.errorGenerator.Generate(); err != nil {
+		return nil, err
+	}
+	return c.baseCMStore.GetClusterMetadata()
+}
+
+func (c *FaultInjectionClusterMetadataStore) SaveClusterMetadata(request *persistence.InternalSaveClusterMetadataRequest) (
+	bool,
+	error,
+) {
+	if err := c.errorGenerator.Generate(); err != nil {
+		return false, err
+	}
+	return c.baseCMStore.SaveClusterMetadata(request)
+}
+
+func (c *FaultInjectionClusterMetadataStore) GetClusterMembers(request *persistence.GetClusterMembersRequest) (
+	*persistence.GetClusterMembersResponse,
+	error,
+) {
+	if err := c.errorGenerator.Generate(); err != nil {
+		return nil, err
+	}
+	return c.baseCMStore.GetClusterMembers(request)
+}
+
+func (c *FaultInjectionClusterMetadataStore) UpsertClusterMembership(request *persistence.UpsertClusterMembershipRequest) error {
+	if err := c.errorGenerator.Generate(); err != nil {
+		return err
+	}
+	return c.baseCMStore.UpsertClusterMembership(request)
+}
+
+func (c *FaultInjectionClusterMetadataStore) PruneClusterMembership(request *persistence.PruneClusterMembershipRequest) error {
+	if err := c.errorGenerator.Generate(); err != nil {
+		return err
+	}
+	return c.baseCMStore.PruneClusterMembership(request)
+}
+
+func NewFaultInjectionMetadataStore(
```

Comment:
should the function return error, does not seem to be possible to return any error?